internal/provisioner/plain/pkg/git/git.go

```diff
@@ -0,0 +1,78 @@
+package git
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+const (
+	defaultDirectory = "./manifests"
+	defaultBranch    = "main"
+)
+
+type CheckoutCmd struct {
+	v1alpha1.GitSource
+}
+
+func NewCheckoutCmd(s v1alpha1.GitSource) CheckoutCmd {
+	return CheckoutCmd{GitSource: s}
+}
+
+func (c CheckoutCmd) String() (string, error) {
+	var checkoutCommand string
+	var repository = c.Repository
+	var directory = c.Directory
+	var branch = c.Ref.Branch
+	var commit = c.Ref.Commit
+	var tag = c.Ref.Tag
+
+	repositoryName, err := c.ValidateRepo()
+	if err != nil {
+		return "", err
+	}
+
+	if directory == "" {
+		directory = defaultDirectory
+	}
+
+	if branch == "" {
+		branch = defaultBranch
+	}
+
+	if commit != "" {
+		checkoutCommand = fmt.Sprintf("git clone %s && cd %s && git checkout %s && cp -r %s/* /manifests",
```

Comment:
How confident are we in extracting the `repositoryName`? Should we instead just create a static dir?

```bash
git clone <repoUrl> repo && cd repo && ...
```

Reply:
That seems a lot safer, was not aware of that option 👍

Reply:
I'm not sure about the use of %q since it seems to add quotation marks where we want to pass unquoted strings in the clone command. We could trim the fields of any whitespace before building the command? Although I'm not sure if the api-server does that when it unmarshals and stores the Bundle to disk.